Mathlib/MeasureTheory/Function/ConvergenceInMeasure.lean

```diff
@@ -219,27 +258,142 @@ theorem TendstoInMeasure.exists_seq_tendsto_ae (hfg : TendstoInMeasure μ f atTo
 
 theorem TendstoInMeasure.exists_seq_tendstoInMeasure_atTop {u : Filter ι} [NeBot u]
     [IsCountablyGenerated u] {f : ι → α → E} {g : α → E} (hfg : TendstoInMeasure μ f u g) :
-    ∃ ns : ℕ → ι, TendstoInMeasure μ (fun n => f (ns n)) atTop g := by
+    ∃ (ns : ℕ → ι) (_ : Tendsto ns atTop u),  TendstoInMeasure μ (fun n => f (ns n)) atTop g := by
   obtain ⟨ns, h_tendsto_ns⟩ : ∃ ns : ℕ → ι, Tendsto ns atTop u := exists_seq_tendsto u
-  exact ⟨ns, fun ε hε => (hfg ε hε).comp h_tendsto_ns⟩
+  exact ⟨ns, h_tendsto_ns, fun ε hε => (hfg ε hε).comp h_tendsto_ns⟩
 
 theorem TendstoInMeasure.exists_seq_tendsto_ae' {u : Filter ι} [NeBot u] [IsCountablyGenerated u]
     {f : ι → α → E} {g : α → E} (hfg : TendstoInMeasure μ f u g) :
-    ∃ ns : ℕ → ι, ∀ᵐ x ∂μ, Tendsto (fun i => f (ns i) x) atTop (𝓝 (g x)) := by
-  obtain ⟨ms, hms⟩ := hfg.exists_seq_tendstoInMeasure_atTop
-  obtain ⟨ns, -, hns⟩ := hms.exists_seq_tendsto_ae
-  exact ⟨ms ∘ ns, hns⟩
+    ∃ (ns : ℕ → ι) (_ : Tendsto ns atTop u),
+    ∀ᵐ x ∂μ, Tendsto (fun i => f (ns i) x) atTop (𝓝 (g x)) := by
+  obtain ⟨ms, hms1, hms2⟩ := hfg.exists_seq_tendstoInMeasure_atTop
+  obtain ⟨ns, hns1, hns2⟩ := hms2.exists_seq_tendsto_ae
+  exact ⟨ms ∘ ns, Tendsto.comp hms1 (StrictMono.tendsto_atTop hns1), hns2⟩
+
+section
+
+/- An auxiliary lemma for a proof by contradiction in exists_seq_tendstoInMeasure_atTop_iff -/
+lemma false_of_tendsto_of_bddBelow_aux (f : ℕ → ℝ≥0) (δ : ℝ) (hδ: (0 : ℝ) < δ)
+    (hf1 : Tendsto f atTop (𝓝 0)) (hf2 : ∀ n, δ ≤ (f n) ) : False := by
+  have h : ∀ x : ℝ≥0, x.toReal = dist x 0 := by
+    intro x
+    rw [NNReal.dist_eq x 0, NNReal.coe_zero, sub_zero, NNReal.abs_eq]
+  simp_rw [h] at hf2
+  apply Metric.false_of_Tendsto_of_bddBelow
+    hδ (Tendsto.comp (NNReal.tendsto_coe'.mpr ⟨Preorder.le_refl 0, fun ⦃_⦄ a ↦ a ⟩) hf1) _
+  refine frequently_atTop'.mpr ?_
+  intro n
+  use n+1
+  simp only [gt_iff_lt, lt_add_iff_pos_right, zero_lt_one, Function.comp_apply, dist_zero_right,
+    Real.norm_eq_abs, NNReal.abs_eq, true_and]
+  have h : ∀ n, dist (f n) 0 = f n := by
+    exact fun n ↦ Eq.symm (Real.ext_cauchy (congrArg Real.cauchy (h (f n))))
+  rw [← h]
+  exact hf2 (n+1)
+
+end
+
+lemma forall_seq_tendstoInMeasure_atTop {u : Filter ι} {v : Filter κ} {f : ι → α → E}
```

Comment:
I don't quite understand this name. I don't see an at top filter here. Maybe the name `TendstoInMeasure.comp_of_tendsto` is a better name?